src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,314 @@
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Ahmed Khalil $
+// $Authors: Ahmed Khalil $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.h>
+
+namespace OpenMS
+{
+  IsotopeLabelingMDVs::IsotopeLabelingMDVs() :
+    DefaultParamHandler("IsotopeLabelingMDVs")
+  {
+  }
+
+  IsotopeLabelingMDVs::~IsotopeLabelingMDVs()
+  {
+  }
+
+  void IsotopeLabelingMDVs::updateMembers_()
+  {
+  }
+  
+  void IsotopeLabelingMDVs::isotopicCorrection(
+    const Feature& normalized_feature,
+    Feature& corrected_feature,
+    const std::vector<std::vector<double>> correction_matrix,
+    const std::string correction_matrix_agent)
+  {
+    // MDV_corrected = correction_matrix_inversed * MDV_observed (normalized_features)
+    
+    std::vector<std::vector<double>> selected_correction_matrix;
+    auto correction_matrix_search = correction_matrices_.find(correction_matrix_agent);
+    
+    if (!correction_matrix_agent.empty() && correction_matrix_search != correction_matrices_.end())
+    {
+      selected_correction_matrix = correction_matrix_search->second ;
+    }
+    else
+    {
+      selected_correction_matrix = correction_matrix;
+    }
+    
+    uint16_t correction_matrix_n = selected_correction_matrix.size();
+    std::vector<std::vector<double>> correction_matrix_inversed(correction_matrix_n, std::vector<double>(correction_matrix_n,0));
```

Comment:
is this one needed or could this be done just via the eigen matrix? so no extra copy would be needed

Reply:
no longer needed